channel/consensus_channel/leader_channel_test.go

```diff
@@ -1,6 +1,7 @@
 package consensus_channel
```

Comment:
General comment: I found this test pretty hard to read. I think that is because test data, helper functions, API calls  and assertions aren't organised very well. 

I'm wondering if a table test or something would help? Something which looks a little more like this when I scan it:

```
0[ ] + Propose(1) = 0[1]
0[1] + Propose(2) = 0[1,2]
0[1,2] + Propose(3) = 0[1,2,3]
0[1,2,3] + UpdateConsensus(2) = 2[3]
2[3] + UpdateConsensus(3) = 3[ ]
```

Here I'm using a notation like `consensusTurnNum.ProposalQueue` (I think you get the idea).

Reply:
Do we have an example of a table test to go off of?

I found [this post](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests), but was not able to find any similar examples of a table test by searching for `range` in our test files:

![Screen Shot 2022-03-28 at 11 49 43 AM](https://user-images.githubusercontent.com/8432675/160466352-dcfc3ff5-5ce0-4f1f-a6d9-c4b07dfeeee8.png)

I will refactor the tests to be along the lines of that post.

Reply:
I think your search there is a malformated. Try this:
![Screenshot 2022-03-29 at 16 59 19](https://user-images.githubusercontent.com/1833419/160654713-c3eee604-c514-4ce5-8c78-eae16cdab95f.png)


I picked out one example: 

https://github.com/statechannels/go-nitro/blob/360812b0883cb2a4b2b53444f47c4d41afddb8f7/types/funds_test.go#L60-L84

I'm not sure how idiomatic it is.